7.scDINO_analysis/2.scDINO_processing/scripts/0.IBP_of_scDINO_profile.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# This notebook performs IBP on the scDINO profile data.
+# We perform the following steps:
+# 1. Load the scDINO profile data.
+# 2. Normalize the data.
+# 3. Feature select the data.
+# 4. Aggregate the data.
+
+# In[1]:
+
+
+import pathlib
+
+import pandas as pd
+from pycytominer import aggregate, feature_select, normalize
+
+# In[2]:
+
+
+scDINO_profile_path = pathlib.Path(
+    "../../1.scDINO_run/outputdir/apoptosis_timelapse/CLS_features/CLS_features_annotated.parquet"
+).resolve(strict=True)
+
+scDINO_normalized_path = pathlib.Path(
+    "../../1.scDINO_run/outputdir/apoptosis_timelapse/CLS_features/CLS_features_annotated_normalized.parquet"
+).resolve()
+scDINO_feature_selected_path = pathlib.Path(
+    "../../1.scDINO_run/outputdir/apoptosis_timelapse/CLS_features/CLS_features_annotated_normalized_feature_selected.parquet"
+).resolve()
+scDINO_aggregated_path = pathlib.Path(
+    "../../1.scDINO_run/outputdir/apoptosis_timelapse/CLS_features/CLS_features_annotated_normalized_feature_selected_aggregated.parquet"
+).resolve()
+
+scDINO_profile = pd.read_parquet(scDINO_profile_path)
+print(f"scDINO profile shape: {scDINO_profile.shape}")
+scDINO_profile.head()
+
+
+# ## Normalization
+
+# In[3]:
+
+
+metadata_columns = [x for x in scDINO_profile.columns if "metadata" in x.lower()]
+features = [x for x in scDINO_profile.columns if x not in metadata_columns]
+normalized_df = normalize(
+    # df with annotated raw merged single cell features
+    profiles=scDINO_profile,
+    # specify samples used as normalization reference (negative control)
+    samples="Metadata_compound == 'Staurosporine' and Metadata_dose == 0.0",
+    # normalization method used
+    method="standardize",
+    features=features,
+    meta_features=metadata_columns,
+)
+
+normalized_df.to_parquet(
+    scDINO_normalized_path,
+    index=False,
+)
+# check to see if the features have been normalized
+print(normalized_df.shape)
+normalized_df.head()
+
+
+# ## Feature Selection
+
+# In[4]:
+
+
+# define operations to be performed on the data
+# list of operations for feature select function to use on input profile
+feature_select_ops = [
+    "variance_threshold",
+    "blocklist",
+    "drop_na_columns",
+    "correlation_threshold",
+]
+manual_block_list = [
+    x
+    for x in normalized_df.columns
+    if "bounding" in x.lower()
+    or "Location_Center_Y" in x.lower()
+    or "Location_Center_X" in x.lower()
+]
+metadata_columns = [x for x in normalized_df.columns if "metadata" in x.lower()]
```

Comment:
consider adding a comment above that you are getting features from the normalized df as this code looks very similar to the scDINO one